crates/miden-objects/src/asset/mod.rs

```diff
@@ -146,7 +146,7 @@ impl Asset {
     }
 
     /// Returns the inner non-fungible asset, or panics if the asset is fungible.
-    pub fn unwrap_non_fungible(&mut self) -> NonFungibleAsset {
```

Comment:
Nit: I'd change the doc comment to have a `# Panics` section similar to other doc comments.